crates/ibc/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -80,6 +82,25 @@ impl ClientState {
         upgrade_path: Vec<String>,
         allow_update: AllowUpdate,
     ) -> Result<ClientState, Error> {
+        if chain_id.as_str().len() > MaxChainIdLen {
+            return Err(Error::chain_id_too_long(
+                chain_id.to_string(),
+                chain_id.as_str().len(),
+                MaxChainIdLen,
+            ));
+        }
+
+        // `TrustThreshold` is guaranteed to be in the range `[0, 1)`, but a `TrustThreshold::ZERO`
+        // value is invalid in this context
+        if trust_level == TrustThreshold::ZERO {
+            return Err(Error::invalid_trust_threshold(
+                "ClientState trust-level cannot be zero".to_string(),
+            ));
+        }
+
+        let _ = TendermintTrustThreshold::new(trust_level.numerator(), trust_level.denominator())
```

Comment:
Note there is currently a [tiny bug](https://github.com/informalsystems/tendermint-rs/issues/1208) in tendermint-rs.

Reply:
Interestingly, the spec also doesn't allow a `TrustThreshold == 1` -> https://github.com/cosmos/ibc/blob/main/spec/client/ics-007-tendermint-client/README.md#client-initialisation